src/main/java/com/ollymonger/dynmap/portals/DynmapPortalsPlugin.java

```diff
@@ -154,12 +157,38 @@ public void onBlockBreak(BlockBreakEvent event) {
                 .findFirst();
 
         portal.ifPresent(p -> {
-            event.getPlayer().sendMessage("You destroyed portal " + p.getPortalId());
+            String portalId = p.getPortalId();
+
+            event.getPlayer().sendMessage("You destroyed portal " + portalId);
+
+            Marker marker = this.portalSet.findMarker(portalId);
+            CircleMarker marker2 = this.portalExclusionSet.findCircleMarker(portalId+"_exclusion");
+            this.deleteMarker(marker);
+            this.deleteCircleMarker(marker2);
 
-            this.registeredPortals.remove(p);
         });
     }
 
+    private void deleteMarker(Marker marker) {
+        try {
+            Method deleteMarker = marker.getClass().getDeclaredMethod("deleteMarker");
+            deleteMarker.setAccessible(true);
+            deleteMarker.invoke(marker);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void deleteCircleMarker(CircleMarker marker2) {
```

Comment:
You don't need to have this method - a `CircleMarker` **is** a `Marker`, so you can just pass it into the standard `deleteMarker` method